docsrc/sample.rst

```diff
@@ -70,28 +71,52 @@ The :ref:`class_cmdstanmodel` class method  ``sample`` returns a ``CmdStanMCMC``
 
-Methods for information about the fit of the model to the data:
+We distinguish between sampler iterations and the resulting sampler
+and parameter values which are written to the Stan CSV output file;
+for the latter, each reported set of estimates constitutes one row's
+worth of data and is called a "draw".  By default, all iterations are
+reported, however, the sampler can be configured to thin the draws.
+In short, the sampler is configuration specifies iterations; the
+results specify draws.
+
+Summarizing and diagnosing the fitted model:
 
 - ``summary()`` - Run CmdStan's `stansummary <https://mc-stan.org/docs/cmdstan-guide/stansummary.html>`__ utility on the sample.
 - ``diagnose()`` - Run CmdStan's `diagnose <https://mc-stan.org/docs/cmdstan-guide/diagnose.html>`__ utility on the sample.
 - ``sampler_diagnostics()`` - Returns the sampler parameters as a map from sampler parameter names to a numpy.ndarray of dimensions draws X chains X 1.
 
-Methods for managing the sample:
+Information about the size and shape of the sample: 
 
-- ``save_csvfiles(dir_name)`` - Move output csvfiles to specified directory.
 - ``chains`` - Number of chains
-- ``num_draws`` - Number of post-warmup draws (i.e., sampling iterations)
-- ``num_warmup_draws`` - Number of warmup draws.
-- ``metric`` - Per chain metric by the HMC sampler.
-- ``stepsize`` - Per chain stepszie used by the HMC sampler.
-- ``sample`` - A 3-D numpy.ndarray which contains all post-warmup draws across all chains arranged as (draws, chains, columns).
-- ``warmup`` - A 3-D numpy.ndarray which contains all warmup draws across all chains arranged as (draws, chains, columns).
+- ``num_draws`` - Number of draws per chain, i.e., thinned iterations.
+- ``num_draws_sampling`` - Number of sampling (post-warmup) draws per chain, i.e., sampling iterations, thinned.  By default, only the post-warmup draws are reported, so that ``num_draws`` == ``num_draws_sampling``.
+- ``num_draws_warmup`` - Number of warmup draws per chain, i.e., thinned warmup iterations.
+
+- ``metric_type`` - Metric type used for adaptation, either ``diag_e``
+  or ``dense_e``, or ``None``, if the Stan program doesn't have any parameters.
+- ``num_params`` - Total number of parameters in the model; the sum of all scalar parameter variables and all elements of all container parameter variables
+
+- ``column_names`` - Column labels for one draw from the sampler.
+- ``sampler_vars_cols`` - Maps the sampler parameter names to output column indices.
+- ``stan_vars_cols`` - Maps the Stan progran variable names to output column indices.
+- ``stan_vars_dims`` - Maps the Stan progran variable names to dimensions; scalar variables have zero dimensions.
 
-Methods for downstream analysis are:
+Contents of the the sample: 
 
+- ``metric`` - Per-chain metric values, either a vector or matrix. 
+- ``step_size`` - Per-chain stepsize.
+  
+- ``draws`` - A numpy.ndarray which contains all across all chains arranged as (raws, chains, columns).
```

Comment:
1. These are the constrained variables?
2. The column names are in `column_names`?
3. I can figure out which columns to find a variable in with `stan_vars_cols`?

Also raws -> draws

Reply:
yes, the constrained variables - that's what the compiled model's `write_array` method does.

Reply:
answers to 2 and 3 also yes and yes - column indexing is 0-based - that's Python